src/resources/Naming.ts

```diff
@@ -104,6 +106,6 @@ export class Naming {
   }
 
   getAuthenticationEmbeddedLamdbaName() {
-    return `${this.apiName}Authorizer`;
+    return `${this.config.name}Authorizer`;
```

Comment:
Is this necessary?

Unless I am missing something, we probably could get away with having a 
`apiLogicalId` that overrides `getLogicalId`.

The rest of the logical ids should be safe. i.e. They might be re-created but they are not stateful. 

That would simplify everything.